File: algorithms/implementation/save-the-prisoner/save-the-prisoner-javascript/js/test_saveThePrisoner.js
import {saveThePrisoner} from '../js/saveThePrisoner.js';
import assert from 'assert';

describe('saveThePrisoner', function() {
    
    it('O número do prisioneiro que recebe o último doce é 2', function() {

        let n = 5;
        let m = 2;
        let s = 1;
        
        let esperado = 2;
        
        let resultado = saveThePrisoner(n, m, s);

        assert.strictEqual(resultado, esperado);
    });
});

describe('saveThePrisoner', function() {
    
    it('O número do prisioneiro que recebe o último doce é 3', function() {

        let n = 5;
        let m = 2;
        let s = 2;
        
        let esperado = 3;
        
        let resultado = saveThePrisoner(n, m, s);

        assert.strictEqual(resultado, esperado);
    });
});

describe('saveThePrisoner', function() {
    
    it('O número do prisioneiro que recebe o último doce é 6', function() {

        let n = 7;
        let m = 19;
        let s = 2;
        
        let esperado = 6;
        
        let resultado = saveThePrisoner(n, m, s);

        assert.strictEqual(resultado, esperado);
    });
});

describe('saveThePrisoner', function() {
    
    it('O número do prisioneiro que recebe o último doce é 3', function() {

        let n = 3;
        let m = 7;
        let s = 3;
        
        let esperado = 3;
        
        let resultado = saveThePrisoner(n, m, s);

        assert.strictEqual(resultado, esperado);
    });
});

File: algorithms/implementation/save-the-prisoner/save-the-prisoner-python/tests/test_saveThePrisoner.py
from saveThePrisoner import saveThePrisoner

class TesteSaveThePrisoner:
    
    def testeCaso1(self):
        
        n = 5
        m = 2
        s = 1
        
        esperado = 2
        
        resultado = saveThePrisoner(n, m, s)
        
        assert resultado == esperado
        
    def testeCaso2(self):
        
        n = 5
        m = 2
        s = 2
        
        esperado = 3
        
        resultado = saveThePrisoner(n, m, s)
        
        assert resultado == esperado
        
    def testeCaso3(self):
        
        n = 7
        m = 19
        s = 2
        
        esperado = 6
        
        resultado = saveThePrisoner(n, m, s)
        
        assert resultado == esperado
        
    def testeCaso4(self):
        
        n = 3
        m = 7
        s = 3
        
        esperado = 3
        
        resultado = saveThePrisoner(n, m, s)
        
        assert resultado == esperado